mount/mount.go

```diff
@@ -39,7 +39,7 @@ var _ ds.Datastore = (*Datastore)(nil)
 
 func (d *Datastore) lookup(key ds.Key) (ds.Datastore, ds.Key, ds.Key) {
 	for _, m := range d.mounts {
-		if m.Prefix.Equal(key) || m.Prefix.IsAncestorOf(key) {
+		if m.Prefix.IsAncestorOf(key) {
```

Comment:
To clarify, is the intent here that if I've mounted `["/", "/foo"]` and then I do a put to `/foo` that the key will end up in the `/` mount instead of the `/foo` mount?

Is there any reason to support this or would we be safer just throwing an error instead?